units/host_allocator.go

```diff
@@ -246,6 +247,39 @@ func (j *hostAllocatorJob) Run(ctx context.Context) {
 	hostQueueRatio := float32(timeToEmpty.Nanoseconds()) / float32(distroQueueInfo.MaxDurationThreshold.Nanoseconds())
 	noSpawnsRatio := float32(timeToEmptyNoSpawns.Nanoseconds()) / float32(distroQueueInfo.MaxDurationThreshold.Nanoseconds())
 
+	const lowRatioThresh = float32(.25)
+	terminateExcess := distro.HostAllocatorSettings.HostsOverallocatedRule == evergreen.HostsOverallocatedTerminate
+	if terminateExcess && hostQueueRatio < lowRatioThresh && len(upHosts) > 0 {
+
+		var killableHosts, newCapTarget int
+		if hostQueueRatio == 0 {
+			killableHosts = len(upHosts)
+			newCapTarget = 0
+		} else {
+			killableHosts = int(float32(len(upHosts)) * (1 - hostQueueRatio))
+			newCapTarget = len(upHosts) - killableHosts
+		}
+
+		if newCapTarget < distro.HostAllocatorSettings.MinimumHosts {
+			newCapTarget = distro.HostAllocatorSettings.MinimumHosts
+		}
+		const lowCountFloor = 5
+		if killableHosts > lowCountFloor {
+
+			drawdownInfo := DrawdownInfo{
+				DistroID:     distro.Id,
+				NewCapTarget: newCapTarget,
+			}
+			queue := j.env.RemoteQueue()
+			err := queue.Put(ctx, NewHostDrawdownJob(j.env, drawdownInfo, utility.RoundPartOfHour(1).Format(TSFormat)))
```

Comment:
I'm concerned that by the time this NewHostDrawdownJob runs, the idle host job might have run and the new cap target might be out of date (this comes from further thinking on our discussion on how these jobs might interact). I guess is it possible to compute this target within the job? I realize the hostQueueRatio might be more difficult to compute there but it could be passed in instead of NewCapTarget) and killableHosts could be redefined at that point. What do you think?

Reply:
I also do worry that there will be too many NewHostDrawdown jobs, given how frequently this job runs.

Reply:
I used to send killableHosts, which has this duplication problem. If, before the drawdown job ran, the hosts were killed by some other process, e.g. the usual idle host termination job, this would be out of date and an overestimate. However, in order to avoid this issue, I send newCapTarget, which is a target # of active hosts, and compute the number of hosts that should be decommissioned based on the running hosts when the drawdown job is run. https://github.com/evergreen-ci/evergreen/blob/88911da4496d15ef54e0b37d6bbd1182c2491f4e/units/distro_overallocated_terminations.go#L91

Reply:
(this is just to say, I think we both had the same idea in the time since I originally submitted the PR)